tests/foundry/lemmas.k

```diff
@@ -0,0 +1,113 @@
+requires "../contracts.k"
+
+module TEST-HELPER-LEMMAS
+  imports ASSERTTEST-CONTRACT
+
+  syntax Bytes ::= testFunction() [function]
+  syntax Set ::= testFunction2(Bytes) [function]
+
+  rule testFunction() => AssertTest.test_assert_true() [simplification]
+  rule testFunction() => #binRuntime(AssertTest) [simplification]
+  rule testFunction2(PGM) => #computeValidJumpDests(PGM) [simplification]
+
+endmodule
+
+module SUM-TO-N-INVARIANT
+
+  imports LOOPSTEST-CONTRACT
+
+    rule N xorInt maxUInt256 <Int S => false
+    requires #rangeUInt(256, S +Int ((N *Int (N +Int 1)) /Int 2))
+    andBool #rangeUInt(256, S)
+    andBool #rangeUInt(256, N)
+    [simplification]
+
+    rule (S +Int N) +Int (((N -Int 1) *Int N) /Int 2) <Int pow256 => true
+    requires #rangeUInt(256, S +Int ((N *Int (N +Int 1)) /Int 2))
+    andBool #rangeUInt(256, S)
+    andBool #rangeUInt(256, N)
+    [simplification]
+
+    rule chop(S +Int N) => S +Int N
+    requires #rangeUInt(256, S +Int ((N *Int (N +Int 1)) /Int 2))
+    andBool #rangeUInt(256, S)
+    andBool #rangeUInt(256, N)
+    [simplification]
+
+    rule chop(N +Int -1) => N +Int -1
+    requires 1 <=Int N andBool N <Int pow256
+    andBool #rangeUInt(256, N)
+    [simplification]
+
+    rule (N *Int (N +Int 1)) /Int 2 <Int 115792089237316195423570985008687907853269984665640564039457584007913129639936 => true
+    requires N <Int 481231938336009023090067544955250113853
+    [simplification]
```

Comment:
This doesn't hold for `N <Int -481231938336009023090067544955250113854`. Why is it needed? The following claim is provable by KEVM master:
```
    claim [sum-to-n-bound]: runLemma ( (N *Int (N +Int 1)) /Int 2 <Int pow256 ) => doneLemma ( true )
      requires -481231938336009023090067544955250113854 <=Int N andBool N <Int 481231938336009023090067544955250113853
```

Reply:
I removed this rule.